operator/apis/metrics/v1alpha1/keptnmetric_types.go

```diff
@@ -77,6 +78,10 @@ func init() {
 	SchemeBuilder.Register(&KeptnMetric{}, &KeptnMetricList{})
 }
 
-func (s KeptnMetric) IsStatusSet() bool {
+func (s *KeptnMetric) IsStatusSet() bool {
 	return s.Status.Value != ""
 }
+
+func (s *KeptnMetric) IsProviderValid(provider string) bool {
```

Comment:
please add a godoc comment for this function with a short description about why the `KeptnMetricProviderName` is not allowed